src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java

```diff
@@ -367,14 +367,18 @@ public UserDTO initiateTrialAccountActivation(@RequestBody String mail) {
     }
 
     @PostMapping(path = "/account/active-trial/finish")
-    public UserDTO finishTrialAccountActivation(@RequestBody KeyAndTermsVM keyAndTermsVM) {
+    public UserDTO finishTrialAccountActivation(@RequestBody KeyAndTermsVM keyAndTermsVM, @RequestParam Boolean newAccountCreation) {
         if (keyAndTermsVM.getReadAndAgreeWithTheTerms() != Boolean.TRUE) {
             throw new AccountResourceException("You have to read and agree with the terms.");
         }
         Optional<UserDetailsDTO> userDetailsDTO = userDetailsService.findOneByTrialActivationKey(keyAndTermsVM.getKey());
         if (userDetailsDTO.isPresent()) {
             Optional<UserDTO> userDTOOptional = userService.finishTrialAccountActivation(keyAndTermsVM.getKey());
             if (userDTOOptional.isPresent()) {
+                // After the user agrees with the terms and is a newly created user, send the OncoKB account is created email.
+                if(newAccountCreation){
+                    mailService.sendCreationEmail(userDTOOptional.get());
+                }
```

Comment:
Not so sure about this, if we are redirecting user to the password reset page right after agreeing to the license term. We might not need this email anymore.